airbyte-integrations/connectors/source-shopify/source_shopify/shopify_graphql/bulk/query.py

```diff
@@ -988,17 +990,60 @@ class InventoryLevel(ShopifyBulkQuery):
         "new_record": "InventoryLevel",
     }
 
+    # quantity related fields and filtering options
+    quantities_names_filter: List[str] = [
+        '"available"',
+        '"incoming"',
+        '"committed"',
+        '"damaged"',
+        '"on_hand"',
+        '"quality_control"',
+        '"reserved"',
+        '"safety_stock"',
+    ]
+    # quantities fields
+    quantities_fields: List[str] = [
+        "id",
+        "name",
+        "quantity",
+        "updatedAt",
+    ]
+
     inventory_levels_fields: List[Field] = [
         "__typename",
         "id",
-        Field(name="available"),
         Field(name="item", fields=[Field(name="id", alias="inventory_item_id")]),
         Field(name="updatedAt"),
     ]
 
+    def _quantities_query(self) -> Query:
+        """
+        Defines the `quantities` nested query.
+        """
+
+        return Query(
+            name="quantities",
+            arguments=[Argument(name="names", value=self.quantities_names_filter)],
```

Comment:
Why do we need this argument? Can't we just get everything? Could we have dataloss if a new quantity.name is added?

Reply:
No, we can't. This is the part of the nested query building process, let's keep this simple.

Reply:
What do you mean by `get just everything?`

Reply:
My understanding by the name of it is that this is a filter meaning that we fetch only the values that are defined in `self.quantities_names_filter`. So effectively, this generate this part of the query:
```
    quantities(names: <list represented by self.quantities_names_filter>) {
      <fields>
    }
```

Can we just query for everything by not passing this filter like this? 
```
    quantities {
      <fields>
    }
```

Reply:
Got your point, we fetch all the values available, there is no wildcard fetch for this, by non-specifying these types. We have to pass all of them as it is now.

Reply:
And is there a way to track new values for this field so that we can update the filter?

Reply:
I think, official docs is the only source of truth, could find any other way of fetching filters, yet.

Reply:
Here is some more info:

```
"errors": [
    {
      "message": "names has an invalid value. Possible values include: available, committed, damaged, incoming, 
    on_hand, quality_control, reserved, safety_stock.",
   ...
   }
]
```

There is no opportunity to pass `*` or `ALL`, or something else. 

The official doc is here: https://shopify.dev/docs/apps/fulfillment/inventory-management-apps#inventory-states

Reply:
I've added the reference link to the code, for quick tips about that, as well.

Reply:
The other possibility I see is an integration tests that query for a value we know does not exist and ensure the error message contains only the possible values we know of. Is this something you think is worth?

Reply:
Created an Issue for the Shopify Comunity with the question and improvement suggestion: https://community.shopify.com/c/graphql-basics-and/graphql-inventorylevels-quantities-issue-with-names-state-filter/m-p/2587605#M13215